sqrt without std:: in Math/GenVector/PxPyPzE4D.h

Hi,

While trying to compile some code which uses ROOT::Math::PxPyPzMVector, I run into
```
In file included from v6-22-06/install/include/Math/Vector4D.h:12:
v6-22-06/install/include/Math/GenVector/PxPyPzE4D.h:154:17: error: call to function 'sqrt' that is neither visible in the template definition nor found by argument-dependent lookup
         return sqrt(mm);
                ^
v6-22-06/install/include/Math/GenVector/PxPyPzM4D.h:108:14: note: in instantiation of member function 'ROOT::Math::PxPyPzE4D<double>::M' requested here
      fM = v.M();

```

In the header files all standard functions should be prefixed with `std::`. I guess this is a remnant from days when there were `using namespace std` statements in the headers...

Cheers,
Antoni

--- a/math/genvector/inc/Math/GenVector/Cartesian2D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian2D.h
@@ -89,7 +89,7 @@ public :
    Scalar X()     const { return fX;}
    Scalar Y()     const { return fY;}
    Scalar Mag2()  const { return fX*fX + fY*fY; }
-   Scalar R() const { return std::sqrt(Mag2()); }
+   Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
    Scalar Phi() const { return (fX == Scalar(0) && fY == Scalar(0)) ? Scalar(0) : atan2(fY, fX); }
 
    /**
@@ -163,8 +163,10 @@ public :
    {
       const Scalar r = v.R(); // re-using this instead of calling v.X() and v.Y()
       // is the speed improvement
-      fX = r * std::cos(v.Phi());
-      fY = r * std::sin(v.Phi());
+      using std::cos;
+      fX = r * cos(v.Phi());
+      using std::sin;
+      fY = r * sin(v.Phi());
    }
    // Technical note:  This works even though only Polar2Dfwd.h is
    // included (and in fact, including Polar2D.h would cause circularity
@@ -175,8 +177,10 @@ public :
    Cartesian2D & operator = (const Polar2D<T2> & v)
    {
       const Scalar r = v.R();
-      fX             = r * std::cos(v.Phi());
-      fY             = r * std::sin(v.Phi());
+      using std::cos;
+      fX             = r * cos(v.Phi());
+      using std::sin;
+      fY             = r * sin(v.Phi());
       return *this;
    }
 

--- a/math/genvector/inc/Math/GenVector/Cartesian2D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian2D.h
@@ -89,7 +89,7 @@ public :
    Scalar X()     const { return fX;}
    Scalar Y()     const { return fY;}
    Scalar Mag2()  const { return fX*fX + fY*fY; }
-   Scalar R() const { return sqrt(Mag2()); }
+   Scalar R() const { return std::sqrt(Mag2()); }
    Scalar Phi() const { return (fX == Scalar(0) && fY == Scalar(0)) ? Scalar(0) : atan2(fY, fX); }
 
    /**

--- a/math/genvector/inc/Math/GenVector/Cartesian2D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian2D.h
@@ -124,8 +124,10 @@ public :
        rotate by an angle
     */
    void Rotate(Scalar angle) {
-      const Scalar s = std::sin(angle);
-      const Scalar c = std::cos(angle);
+      using std::sin;
+      const Scalar s = sin(angle);
+      using std::cos;
+      const Scalar c = cos(angle);
       SetCoordinates(c * fX - s * fY, s * fX + c * fY);
    }
 

--- a/math/genvector/inc/Math/GenVector/Cartesian2D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian2D.h
@@ -124,8 +124,8 @@ public :
        rotate by an angle
     */
    void Rotate(Scalar angle) {
-      const Scalar s = sin(angle);
-      const Scalar c = cos(angle);
+      const Scalar s = std::sin(angle);
+      const Scalar c = std::cos(angle);
       SetCoordinates(c * fX - s * fY, s * fX + c * fY);
    }
 
@@ -175,8 +175,8 @@ public :
    Cartesian2D & operator = (const Polar2D<T2> & v)
    {
       const Scalar r = v.R();
-      fX             = r * cos(v.Phi());
-      fY             = r * sin(v.Phi());
+      fX             = r * std::cos(v.Phi());
+      fY             = r * std::sin(v.Phi());
       return *this;
    }
 

--- a/math/genvector/inc/Math/GenVector/Cartesian2D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian2D.h
@@ -90,7 +90,7 @@ public :
    Scalar Y()     const { return fY;}
    Scalar Mag2()  const { return fX*fX + fY*fY; }
    Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
-   Scalar Phi() const { return (fX == Scalar(0) && fY == Scalar(0)) ? Scalar(0) : atan2(fY, fX); }
+   Scalar Phi() const { using std::atan2; return (fX == Scalar(0) && fY == Scalar(0)) ? Scalar(0) : atan2(fY, fX); }
 
    /**
        set the x coordinate value keeping y constant

--- a/math/genvector/inc/Math/GenVector/Cartesian3D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian3D.h
@@ -111,8 +111,8 @@ public :
    Scalar Perp2() const { return fX*fX + fY*fY ;}
    Scalar Rho() const { using std::sqrt; return sqrt(Perp2()); }
    Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
-   Scalar Theta() const { return atan2(Rho(), Z()); }
-   Scalar Phi() const { return atan2(fY, fX); }
+   Scalar Theta() const { using std::atan2; return atan2(Rho(), Z()); }
+   Scalar Phi() const { using std::atan2; return atan2(fY, fX); }
 
    // pseudorapidity
    Scalar Eta() const {

--- a/math/genvector/inc/Math/GenVector/Cartesian3D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian3D.h
@@ -207,8 +207,8 @@ public :
    Cartesian3D & operator = (const Polar3D<T2> & v)
    {
       const T rho = v.Rho();
-      fX          = rho * cos(v.Phi());
-      fY          = rho * sin(v.Phi());
+      fX          = rho * std::cos(v.Phi());
+      fY          = rho * std::sin(v.Phi());
       fZ = v.Z();
       return *this;
    }

--- a/math/genvector/inc/Math/GenVector/Cartesian3D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian3D.h
@@ -109,7 +109,7 @@ public :
    Scalar Z()     const { return fZ;}
    Scalar Mag2()  const { return fX*fX + fY*fY + fZ*fZ;}
    Scalar Perp2() const { return fX*fX + fY*fY ;}
-   Scalar Rho() const { return std::sqrt(Perp2()); }
+   Scalar Rho() const { using std::sqrt; return sqrt(Perp2()); }
    Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
    Scalar Theta() const { return atan2(Rho(), Z()); }
    Scalar Phi() const { return atan2(fY, fX); }
@@ -207,8 +207,10 @@ public :
    Cartesian3D & operator = (const Polar3D<T2> & v)
    {
       const T rho = v.Rho();
-      fX          = rho * std::cos(v.Phi());
-      fY          = rho * std::sin(v.Phi());
+      using std::cos;
+      fX          = rho * cos(v.Phi());
+      using std::sin;
+      fY          = rho * sin(v.Phi());
       fZ = v.Z();
       return *this;
    }

--- a/math/genvector/inc/Math/GenVector/Cartesian3D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian3D.h
@@ -109,8 +109,8 @@ public :
    Scalar Z()     const { return fZ;}
    Scalar Mag2()  const { return fX*fX + fY*fY + fZ*fZ;}
    Scalar Perp2() const { return fX*fX + fY*fY ;}
-   Scalar Rho() const { return sqrt(Perp2()); }
-   Scalar R() const { return sqrt(Mag2()); }
+   Scalar Rho() const { return std::sqrt(Perp2()); }
+   Scalar R() const { return std::sqrt(Mag2()); }
    Scalar Theta() const { return atan2(Rho(), Z()); }
    Scalar Phi() const { return atan2(fY, fX); }
 

--- a/math/genvector/inc/Math/GenVector/Cartesian3D.h
+++ b/math/genvector/inc/Math/GenVector/Cartesian3D.h
@@ -110,7 +110,7 @@ public :
    Scalar Mag2()  const { return fX*fX + fY*fY + fZ*fZ;}
    Scalar Perp2() const { return fX*fX + fY*fY ;}
    Scalar Rho() const { return std::sqrt(Perp2()); }
-   Scalar R() const { return std::sqrt(Mag2()); }
+   Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
    Scalar Theta() const { return atan2(Rho(), Z()); }
    Scalar Phi() const { return atan2(fY, fX); }
 

--- a/math/genvector/inc/Math/GenVector/Cylindrical3D.h
+++ b/math/genvector/inc/Math/GenVector/Cylindrical3D.h
@@ -109,7 +109,8 @@ private:
    inline static Scalar pi() { return Scalar(M_PI); }
    inline void          Restrict()
    {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 public:
 
@@ -118,12 +119,11 @@ public:
    Scalar Rho()   const { return fRho; }
    Scalar Z()     const { return fZ;   }
    Scalar Phi()   const { return fPhi; }
-
-   Scalar X() const { return fRho * std::cos(fPhi); }
-   Scalar Y() const { return fRho * std::sin(fPhi); }
+   Scalar X() const { using std::cos; return fRho * cos(fPhi); }
+   Scalar Y() const { using std::sin; return fRho * sin(fPhi); }
 
    Scalar Mag2()  const { return fRho*fRho + fZ*fZ;   }
-   Scalar R() const { return std::sqrt(Mag2()); }
+   Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
    Scalar Perp2() const { return fRho*fRho;           }
    Scalar Theta() const { return (fRho == Scalar(0) && fZ == Scalar(0)) ? Scalar(0) : atan2(fRho, fZ); }
 

--- a/math/genvector/inc/Math/GenVector/Cylindrical3D.h
+++ b/math/genvector/inc/Math/GenVector/Cylindrical3D.h
@@ -119,8 +119,8 @@ public:
    Scalar Z()     const { return fZ;   }
    Scalar Phi()   const { return fPhi; }
 
-   Scalar X() const { return fRho * cos(fPhi); }
-   Scalar Y() const { return fRho * sin(fPhi); }
+   Scalar X() const { return fRho * std::cos(fPhi); }
+   Scalar Y() const { return fRho * std::sin(fPhi); }
 
    Scalar Mag2()  const { return fRho*fRho + fZ*fZ;   }
    Scalar R() const { return std::sqrt(Mag2()); }

--- a/math/genvector/inc/Math/GenVector/Cylindrical3D.h
+++ b/math/genvector/inc/Math/GenVector/Cylindrical3D.h
@@ -123,7 +123,7 @@ public:
    Scalar Y() const { return fRho * sin(fPhi); }
 
    Scalar Mag2()  const { return fRho*fRho + fZ*fZ;   }
-   Scalar R() const { return sqrt(Mag2()); }
+   Scalar R() const { return std::sqrt(Mag2()); }
    Scalar Perp2() const { return fRho*fRho;           }
    Scalar Theta() const { return (fRho == Scalar(0) && fZ == Scalar(0)) ? Scalar(0) : atan2(fRho, fZ); }
 

--- a/math/genvector/inc/Math/GenVector/Cylindrical3D.h
+++ b/math/genvector/inc/Math/GenVector/Cylindrical3D.h
@@ -109,7 +109,7 @@ private:
    inline static Scalar pi() { return Scalar(M_PI); }
    inline void          Restrict()
    {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 public:
 

--- a/math/genvector/inc/Math/GenVector/Cylindrical3D.h
+++ b/math/genvector/inc/Math/GenVector/Cylindrical3D.h
@@ -125,7 +125,7 @@ public:
    Scalar Mag2()  const { return fRho*fRho + fZ*fZ;   }
    Scalar R() const { using std::sqrt; return sqrt(Mag2()); }
    Scalar Perp2() const { return fRho*fRho;           }
-   Scalar Theta() const { return (fRho == Scalar(0) && fZ == Scalar(0)) ? Scalar(0) : atan2(fRho, fZ); }
+   Scalar Theta() const { using std::atan2; return (fRho == Scalar(0) && fZ == Scalar(0)) ? Scalar(0) : atan2(fRho, fZ); }
 
    // pseudorapidity - use same implementation as in Cartesian3D
    Scalar Eta() const {

--- a/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
+++ b/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
@@ -123,7 +123,8 @@ public :
 private:
    inline static Scalar pi() { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
       return;
    }
 public:
@@ -133,15 +134,17 @@ public:
    T Rho()   const { return fRho; }
    T Eta()   const { return fEta; }
    T Phi()   const { return fPhi; }
-   T X() const { return fRho * std::cos(fPhi); }
-   T Y() const { return fRho * std::sin(fPhi); }
+   T X() const { using std::cos; return fRho * cos(fPhi); }
+   T Y() const { using std::sin; return fRho * sin(fPhi); }
    T Z() const
    {
-      return fRho > 0 ? fRho * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<T>() : fEta + etaMax<T>();
+      using std::sinh;
+      return fRho > 0 ? fRho * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<T>() : fEta + etaMax<T>();
    }
    T R() const
    {
-      return fRho > 0 ? fRho * std::cosh(fEta)
+      using std::cosh;
+      return fRho > 0 ? fRho * cosh(fEta)
                       : fEta > etaMax<T>() ? fEta - etaMax<T>() : fEta < -etaMax<T>() ? -fEta - etaMax<T>() : 0;
    }
    T Mag2() const
@@ -150,7 +153,7 @@ public:
       return r * r;
    }
    T Perp2() const { return fRho*fRho;            }
-   T Theta() const { return fRho > 0 ? 2 * std::atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }
+   T Theta() const { using std::atan; return fRho > 0 ? 2 * atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }
 
    // setters (only for data members)
 

--- a/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
+++ b/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
@@ -123,7 +123,7 @@ public :
 private:
    inline static Scalar pi() { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
       return;
    }
 public:

--- a/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
+++ b/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
@@ -68,6 +68,7 @@ public :
   explicit CylindricalEta3D( const CoordSystem & v ) :
      fRho(v.Rho() ),  fEta(v.Eta() ),  fPhi(v.Phi() )
   {
+     using std::log; 
      static Scalar bigEta = Scalar(-0.3) * log(std::numeric_limits<Scalar>::epsilon());
      if (std::fabs(fEta) > bigEta) {
         // This gives a small absolute adjustment in rho,

--- a/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
+++ b/math/genvector/inc/Math/GenVector/CylindricalEta3D.h
@@ -133,15 +133,15 @@ public:
    T Rho()   const { return fRho; }
    T Eta()   const { return fEta; }
    T Phi()   const { return fPhi; }
-   T X() const { return fRho * cos(fPhi); }
-   T Y() const { return fRho * sin(fPhi); }
+   T X() const { return fRho * std::cos(fPhi); }
+   T Y() const { return fRho * std::sin(fPhi); }
    T Z() const
    {
-      return fRho > 0 ? fRho * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<T>() : fEta + etaMax<T>();
+      return fRho > 0 ? fRho * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<T>() : fEta + etaMax<T>();
    }
    T R() const
    {
-      return fRho > 0 ? fRho * cosh(fEta)
+      return fRho > 0 ? fRho * std::cosh(fEta)
                       : fEta > etaMax<T>() ? fEta - etaMax<T>() : fEta < -etaMax<T>() ? -fEta - etaMax<T>() : 0;
    }
    T Mag2() const
@@ -150,7 +150,7 @@ public:
       return r * r;
    }
    T Perp2() const { return fRho*fRho;            }
-   T Theta() const { return fRho > 0 ? 2 * atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }
+   T Theta() const { return fRho > 0 ? 2 * std::atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }
 
    // setters (only for data members)
 

--- a/math/genvector/inc/Math/GenVector/LorentzVector.h
+++ b/math/genvector/inc/Math/GenVector/LorentzVector.h
@@ -614,7 +614,8 @@ More details about the GenVector package can be found \ref GenVector "here".
           else if ( t2 == v2 ) {
              GenVector::Throw ("LorentzVector::Gamma() - gamma computed for a lightlike LorentzVector. Infinite result");
           }
-          return Scalar(1) / std::sqrt(Scalar(1) - v2 / t2);
+          using std::sqrt;
+          return Scalar(1) / sqrt(Scalar(1) - v2 / t2);
        } /* gamma */
 
 

--- a/math/genvector/inc/Math/GenVector/LorentzVector.h
+++ b/math/genvector/inc/Math/GenVector/LorentzVector.h
@@ -490,6 +490,7 @@ More details about the GenVector package can be found \ref GenVector "here".
           //        We should then move the code to a .cpp file.
           const Scalar ee  = E();
           const Scalar ppz = Pz();
+          using std::log;
           return Scalar(0.5) * log((ee + ppz) / (ee - ppz));
        }
 
@@ -501,6 +502,7 @@ More details about the GenVector package can be found \ref GenVector "here".
           //        mechanism or at least load a NAN if not.
           const Scalar ee = E();
           const Scalar pp = P();
+          using std::log;
           return Scalar(0.5) * log((ee + pp) / (ee - pp));
        }
 

--- a/math/genvector/inc/Math/GenVector/LorentzVector.h
+++ b/math/genvector/inc/Math/GenVector/LorentzVector.h
@@ -614,7 +614,7 @@ More details about the GenVector package can be found \ref GenVector "here".
           else if ( t2 == v2 ) {
              GenVector::Throw ("LorentzVector::Gamma() - gamma computed for a lightlike LorentzVector. Infinite result");
           }
-          return Scalar(1) / sqrt(Scalar(1) - v2 / t2);
+          return Scalar(1) / std::sqrt(Scalar(1) - v2 / t2);
        } /* gamma */
 
 
@@ -769,6 +769,3 @@ std::string printValue(const ROOT::Math::LorentzVector<CoordSystem> *v)
 #endif
 
 //#include "Math/GenVector/LorentzVectorOperations.h"
-
-
-

--- a/math/genvector/inc/Math/GenVector/Plane3D.h
+++ b/math/genvector/inc/Math/GenVector/Plane3D.h
@@ -216,7 +216,7 @@ protected:
    void Normalize()
    {
       // normalize the plane
-      const SCALAR s = sqrt(fA * fA + fB * fB + fC * fC);
+      const SCALAR s = std::sqrt(fA * fA + fB * fB + fC * fC);
       // what to do if s = 0 ?
       if (s == SCALAR(0)) {
          fD = SCALAR(0);
@@ -236,7 +236,7 @@ protected:
    void Normalize()
    {
       // normalize the plane
-      SCALAR s = sqrt(fA * fA + fB * fB + fC * fC);
+      SCALAR s = std::sqrt(fA * fA + fB * fB + fC * fC);
       // what to do if s = 0 ?
       const auto m = (s == SCALAR(0));
       // set zero entries to 1 in the vector to avoid /0 later on
@@ -308,7 +308,3 @@ private:
 
 
 #endif
-
-
-
-

--- a/math/genvector/inc/Math/GenVector/Plane3D.h
+++ b/math/genvector/inc/Math/GenVector/Plane3D.h
@@ -216,7 +216,8 @@ protected:
    void Normalize()
    {
       // normalize the plane
-      const SCALAR s = std::sqrt(fA * fA + fB * fB + fC * fC);
+      using std::sqrt;
+      const SCALAR s = sqrt(fA * fA + fB * fB + fC * fC);
       // what to do if s = 0 ?
       if (s == SCALAR(0)) {
          fD = SCALAR(0);

--- a/math/genvector/inc/Math/GenVector/Plane3D.h
+++ b/math/genvector/inc/Math/GenVector/Plane3D.h
@@ -236,7 +236,8 @@ protected:
    void Normalize()
    {
       // normalize the plane
-      SCALAR s = std::sqrt(fA * fA + fB * fB + fC * fC);
+      using std::sqrt;
+      SCALAR s = sqrt(fA * fA + fB * fB + fC * fC);
       // what to do if s = 0 ?
       const auto m = (s == SCALAR(0));
       // set zero entries to 1 in the vector to avoid /0 later on

--- a/math/genvector/inc/Math/GenVector/Polar2D.h
+++ b/math/genvector/inc/Math/GenVector/Polar2D.h
@@ -97,8 +97,8 @@ public :
 
    Scalar R()     const { return fR;}
    Scalar Phi()   const { return fPhi; }
-   Scalar X() const { return fR * cos(fPhi); }
-   Scalar Y() const { return fR * sin(fPhi); }
+   Scalar X() const { return fR * std::cos(fPhi); }
+   Scalar Y() const { return fR * std::sin(fPhi); }
    Scalar Mag2()  const { return fR*fR;}
 
 

--- a/math/genvector/inc/Math/GenVector/Polar2D.h
+++ b/math/genvector/inc/Math/GenVector/Polar2D.h
@@ -97,8 +97,8 @@ public :
 
    Scalar R()     const { return fR;}
    Scalar Phi()   const { return fPhi; }
-   Scalar X() const { return fR * std::cos(fPhi); }
-   Scalar Y() const { return fR * std::sin(fPhi); }
+   Scalar X() const { using std::cos; return fR * cos(fPhi); }
+   Scalar Y() const { using std::sin; return fR * sin(fPhi); }
    Scalar Mag2()  const { return fR*fR;}
 
 
@@ -134,7 +134,8 @@ private:
       restrict abgle hi to be between -PI and PI
     */
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 
 public:

--- a/math/genvector/inc/Math/GenVector/Polar2D.h
+++ b/math/genvector/inc/Math/GenVector/Polar2D.h
@@ -134,7 +134,7 @@ private:
       restrict abgle hi to be between -PI and PI
     */
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 
 public:

--- a/math/genvector/inc/Math/GenVector/Polar3D.h
+++ b/math/genvector/inc/Math/GenVector/Polar3D.h
@@ -110,10 +110,10 @@ public :
    Scalar R()     const { return fR;}
    Scalar Phi()   const { return fPhi; }
    Scalar Theta() const { return fTheta; }
-   Scalar Rho() const { return fR * std::sin(fTheta); }
-   Scalar X() const { return Rho() * std::cos(fPhi); }
-   Scalar Y() const { return Rho() * std::sin(fPhi); }
-   Scalar Z() const { return fR * std::cos(fTheta); }
+   Scalar Rho() const { using std::sin; return fR * sin(fTheta); }
+   Scalar X() const { using std::cos; return Rho() * cos(fPhi); }
+   Scalar Y() const { using std::sin; return Rho() * sin(fPhi); }
+   Scalar Z() const { using std::cos; return fR * cos(fTheta); }
    Scalar Mag2()  const { return fR*fR;}
    Scalar Perp2() const { return Rho() * Rho(); }
 
@@ -157,7 +157,8 @@ public :
 private:
    inline static Scalar pi()  { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 
 public:

--- a/math/genvector/inc/Math/GenVector/Polar3D.h
+++ b/math/genvector/inc/Math/GenVector/Polar3D.h
@@ -110,10 +110,10 @@ public :
    Scalar R()     const { return fR;}
    Scalar Phi()   const { return fPhi; }
    Scalar Theta() const { return fTheta; }
-   Scalar Rho() const { return fR * sin(fTheta); }
-   Scalar X() const { return Rho() * cos(fPhi); }
-   Scalar Y() const { return Rho() * sin(fPhi); }
-   Scalar Z() const { return fR * cos(fTheta); }
+   Scalar Rho() const { return fR * std::sin(fTheta); }
+   Scalar X() const { return Rho() * std::cos(fPhi); }
+   Scalar Y() const { return Rho() * std::sin(fPhi); }
+   Scalar Z() const { return fR * std::cos(fTheta); }
    Scalar Mag2()  const { return fR*fR;}
    Scalar Perp2() const { return Rho() * Rho(); }
 

--- a/math/genvector/inc/Math/GenVector/Polar3D.h
+++ b/math/genvector/inc/Math/GenVector/Polar3D.h
@@ -157,7 +157,7 @@ public :
 private:
    inline static Scalar pi()  { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 
 public:

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
@@ -237,7 +237,7 @@ public :
 private:
    inline static Scalar pi() { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 public:
 

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
@@ -181,11 +181,11 @@ public :
    Scalar M() const    {
       const Scalar mm = M2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw ("PtEtaPhiE4D::M() - Tachyonic:\n"
                            "    Pt and Eta give P such that P^2 > E^2, so the mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
    Scalar Mag() const    { return M(); }
@@ -207,11 +207,11 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw ("PtEtaPhiE4D::Mt() - Tachyonic:\n"
                            "    Pt and Eta give Pz such that Pz^2 > E^2, so the mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
 
@@ -421,4 +421,3 @@ inline void PtEtaPhiE4D<ScalarType>::SetM(Scalar m) {
 
 
 #endif // ROOT_Math_GenVector_PtEtaPhiE4D
-

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
@@ -137,12 +137,12 @@ public :
 
    // other coordinate representation
 
-   Scalar Px() const { return fPt * cos(fPhi); }
+   Scalar Px() const { return fPt * std::cos(fPhi); }
    Scalar X () const { return Px();         }
-   Scalar Py() const { return fPt * sin(fPhi); }
+   Scalar Py() const { return fPt * std::sin(fPhi); }
    Scalar Y () const { return Py();         }
    Scalar Pz() const {
-      return fPt > 0 ? fPt * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
+      return fPt > 0 ? fPt * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
    }
    Scalar Z () const { return Pz(); }
 
@@ -150,7 +150,7 @@ public :
        magnitude of momentum
    */
    Scalar P() const {
-      return fPt > 0 ? fPt * cosh(fEta)
+      return fPt > 0 ? fPt * std::cosh(fEta)
                      : fEta > etaMax<Scalar>() ? fEta - etaMax<Scalar>()
                                                : fEta < -etaMax<Scalar>() ? -fEta - etaMax<Scalar>() : 0;
    }
@@ -222,7 +222,7 @@ public :
       transverse energy
    */
    Scalar Et() const {
-      return fE / cosh(fEta); // faster using eta
+      return fE / std::cosh(fEta); // faster using eta
    }
 
    /**
@@ -244,7 +244,7 @@ public:
    /**
       polar angle
    */
-   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
+   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * std::atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
 
    // --------- Set Coordinates of this system  ---------------
 

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiE4D.h
@@ -137,12 +137,13 @@ public :
 
    // other coordinate representation
 
-   Scalar Px() const { return fPt * std::cos(fPhi); }
+   Scalar Px() const { using std::cos; return fPt * cos(fPhi); }
    Scalar X () const { return Px();         }
-   Scalar Py() const { return fPt * std::sin(fPhi); }
+   Scalar Py() const { using std::sin; return fPt * sin(fPhi); }
    Scalar Y () const { return Py();         }
    Scalar Pz() const {
-      return fPt > 0 ? fPt * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
+      using std:: sinh;
+      return fPt > 0 ? fPt * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
    }
    Scalar Z () const { return Pz(); }
 
@@ -150,7 +151,8 @@ public :
        magnitude of momentum
    */
    Scalar P() const {
-      return fPt > 0 ? fPt * std::cosh(fEta)
+     using std::cosh;
+      return fPt > 0 ? fPt * cosh(fEta)
                      : fEta > etaMax<Scalar>() ? fEta - etaMax<Scalar>()
                                                : fEta < -etaMax<Scalar>() ? -fEta - etaMax<Scalar>() : 0;
    }
@@ -181,11 +183,13 @@ public :
    Scalar M() const    {
       const Scalar mm = M2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw ("PtEtaPhiE4D::M() - Tachyonic:\n"
                            "    Pt and Eta give P such that P^2 > E^2, so the mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
    Scalar Mag() const    { return M(); }
@@ -207,11 +211,13 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw ("PtEtaPhiE4D::Mt() - Tachyonic:\n"
                            "    Pt and Eta give Pz such that Pz^2 > E^2, so the mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
 
@@ -222,7 +228,8 @@ public :
       transverse energy
    */
    Scalar Et() const {
-      return fE / std::cosh(fEta); // faster using eta
+      using std::cosh;
+      return fE / cosh(fEta); // faster using eta
    }
 
    /**
@@ -237,14 +244,15 @@ public :
 private:
    inline static Scalar pi() { return M_PI; }
    inline void Restrict() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
 public:
 
    /**
       polar angle
    */
-   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * std::atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
+   Scalar Theta() const { using std::atan; return (fPt > 0 ? Scalar(2) * atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
 
    // --------- Set Coordinates of this system  ---------------
 

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
@@ -190,7 +190,7 @@ public :
    /**
        Energy (timelike component of momentum-energy 4-vector)
    */
-   Scalar E() const { return sqrt(E2()); }
+   Scalar E() const { return std::sqrt(E2()); }
 
    Scalar T()   const { return E();  }
 
@@ -220,11 +220,11 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw  ("PtEtaPhiM4D::Mt() - Tachyonic:\n"
                             "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
 
@@ -437,4 +437,3 @@ void PtEtaPhiM4D<ScalarType>::SetE(Scalar energy) {
 
 
 #endif // ROOT_Math_GenVector_PtEtaPhiM4D
-

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
@@ -150,12 +150,12 @@ public :
 
    // other coordinate representation
 
-   Scalar Px() const { return fPt * cos(fPhi); }
+   Scalar Px() const { return fPt * std::cos(fPhi); }
    Scalar X () const { return Px();         }
-   Scalar Py() const { return fPt * sin(fPhi); }
+   Scalar Py() const { return fPt * std::sin(fPhi); }
    Scalar Y () const { return Py();         }
    Scalar Pz() const {
-      return fPt > 0 ? fPt * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
+      return fPt > 0 ? fPt * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
    }
    Scalar Z () const { return Pz(); }
 
@@ -163,7 +163,7 @@ public :
        magnitude of momentum
    */
    Scalar P() const {
-      return fPt > 0 ? fPt * cosh(fEta)
+      return fPt > 0 ? fPt * std::cosh(fEta)
                      : fEta > etaMax<Scalar>() ? fEta - etaMax<Scalar>()
                                                : fEta < -etaMax<Scalar>() ? -fEta - etaMax<Scalar>() : 0;
    }
@@ -233,18 +233,18 @@ public :
    */
    Scalar Et2() const {
       // a bit faster than et * et
-      return 2. * E2() / (cosh(2 * fEta) + 1);
+      return 2. * E2() / (std::cosh(2 * fEta) + 1);
    }
 
    /**
       transverse energy
    */
-   Scalar Et() const { return E() / cosh(fEta); }
+   Scalar Et() const { return E() / std::cosh(fEta); }
 
 private:
    inline static Scalar pi() { return M_PI; }
    inline void RestrictPhi() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
    // restrict the value of negative mass to avoid unphysical negative E2 values
    // M2 must be less than P2 for the tachionic particles - otherwise use positive values
@@ -262,7 +262,7 @@ public:
    /**
       polar angle
    */
-   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
+   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * std::atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
 
    // --------- Set Coordinates of this system  ---------------
 

--- a/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
+++ b/math/genvector/inc/Math/GenVector/PtEtaPhiM4D.h
@@ -150,12 +150,13 @@ public :
 
    // other coordinate representation
 
-   Scalar Px() const { return fPt * std::cos(fPhi); }
+   Scalar Px() const { using std::cos; return fPt * cos(fPhi); }
    Scalar X () const { return Px();         }
-   Scalar Py() const { return fPt * std::sin(fPhi); }
+   Scalar Py() const { using std::sin; return fPt * sin(fPhi); }
    Scalar Y () const { return Py();         }
    Scalar Pz() const {
-      return fPt > 0 ? fPt * std::sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
+      using std::sinh;
+      return fPt > 0 ? fPt * sinh(fEta) : fEta == 0 ? 0 : fEta > 0 ? fEta - etaMax<Scalar>() : fEta + etaMax<Scalar>();
    }
    Scalar Z () const { return Pz(); }
 
@@ -163,7 +164,8 @@ public :
        magnitude of momentum
    */
    Scalar P() const {
-      return fPt > 0 ? fPt * std::cosh(fEta)
+      using std::cosh;
+      return fPt > 0 ? fPt * cosh(fEta)
                      : fEta > etaMax<Scalar>() ? fEta - etaMax<Scalar>()
                                                : fEta < -etaMax<Scalar>() ? -fEta - etaMax<Scalar>() : 0;
    }
@@ -190,7 +192,7 @@ public :
    /**
        Energy (timelike component of momentum-energy 4-vector)
    */
-   Scalar E() const { return std::sqrt(E2()); }
+   Scalar E() const { using std::sqrt; return sqrt(E2()); }
 
    Scalar T()   const { return E();  }
 
@@ -220,11 +222,13 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw  ("PtEtaPhiM4D::Mt() - Tachyonic:\n"
                             "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
 
@@ -233,18 +237,20 @@ public :
    */
    Scalar Et2() const {
       // a bit faster than et * et
-      return 2. * E2() / (std::cosh(2 * fEta) + 1);
+      using std::cosh;
+      return 2. * E2() / (cosh(2 * fEta) + 1);
    }
 
    /**
       transverse energy
    */
-   Scalar Et() const { return E() / std::cosh(fEta); }
+   Scalar Et() const { using std::cosh; return E() / cosh(fEta); }
 
 private:
    inline static Scalar pi() { return M_PI; }
    inline void RestrictPhi() {
-      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - std::floor(fPhi / (2 * pi()) + .5) * 2 * pi();
+      using std::floor;
+      if (fPhi <= -pi() || fPhi > pi()) fPhi = fPhi - floor(fPhi / (2 * pi()) + .5) * 2 * pi();
    }
    // restrict the value of negative mass to avoid unphysical negative E2 values
    // M2 must be less than P2 for the tachionic particles - otherwise use positive values
@@ -262,7 +268,7 @@ public:
    /**
       polar angle
    */
-   Scalar Theta() const { return (fPt > 0 ? Scalar(2) * std::atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
+   Scalar Theta() const { using std::atan; return (fPt > 0 ? Scalar(2) * atan(exp(-fEta)) : fEta >= 0 ? 0 : pi()); }
 
    // --------- Set Coordinates of this system  ---------------
 

--- a/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
@@ -135,7 +135,7 @@ public :
    /**
       magnitude of spatial components (magnitude of 3-momentum)
    */
-   Scalar P() const { return std::sqrt(P2()); }
+   Scalar P() const { using std::sqrt; return sqrt(P2()); }
    Scalar R() const { return P(); }
 
    /**
@@ -151,11 +151,13 @@ public :
    {
       const Scalar mm = M2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzE4D::M() - Tachyonic:\n"
                    "    P^2 > E^2 so the mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
    Scalar Mag() const    { return M(); }
@@ -169,7 +171,7 @@ public :
    /**
       Transverse spatial component (P_perp or rho)
    */
-   Scalar Pt() const { return std::sqrt(Perp2()); }
+   Scalar Pt() const { using std::sqrt; return sqrt(Perp2()); }
    Scalar Perp() const { return Pt();}
    Scalar Rho()  const { return Pt();}
 
@@ -184,11 +186,13 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzE4D::Mt() - Tachyonic:\n"
                            "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
 
@@ -206,7 +210,8 @@ public :
    */
    Scalar Et() const {
       const Scalar etet = Et2();
-      return fT < 0.0 ? -std::sqrt(etet) : std::sqrt(etet);
+      using std::sqrt;
+      return fT < 0.0 ? -sqrt(etet) : sqrt(etet);
    }
 
    /**

--- a/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
@@ -135,7 +135,7 @@ public :
    /**
       magnitude of spatial components (magnitude of 3-momentum)
    */
-   Scalar P() const { return sqrt(P2()); }
+   Scalar P() const { return std::sqrt(P2()); }
    Scalar R() const { return P(); }
 
    /**
@@ -151,11 +151,11 @@ public :
    {
       const Scalar mm = M2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzE4D::M() - Tachyonic:\n"
                    "    P^2 > E^2 so the mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
    Scalar Mag() const    { return M(); }
@@ -169,7 +169,7 @@ public :
    /**
       Transverse spatial component (P_perp or rho)
    */
-   Scalar Pt() const { return sqrt(Perp2()); }
+   Scalar Pt() const { return std::sqrt(Perp2()); }
    Scalar Perp() const { return Pt();}
    Scalar Rho()  const { return Pt();}
 
@@ -184,11 +184,11 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzE4D::Mt() - Tachyonic:\n"
                            "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
 
@@ -206,7 +206,7 @@ public :
    */
    Scalar Et() const {
       const Scalar etet = Et2();
-      return fT < 0.0 ? -sqrt(etet) : sqrt(etet);
+      return fT < 0.0 ? -std::sqrt(etet) : std::sqrt(etet);
    }
 
    /**

--- a/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzE4D.h
@@ -217,12 +217,12 @@ public :
    /**
       azimuthal angle
    */
-   Scalar Phi() const { return (fX == 0.0 && fY == 0.0) ? 0 : atan2(fY, fX); }
+   Scalar Phi() const { using std::atan2; return (fX == 0.0 && fY == 0.0) ? 0 : atan2(fY, fX); }
 
    /**
       polar angle
    */
-   Scalar Theta() const { return (fX == 0.0 && fY == 0.0 && fZ == 0.0) ? 0 : atan2(Pt(), fZ); }
+   Scalar Theta() const { using std::atan2; return (fX == 0.0 && fY == 0.0 && fZ == 0.0) ? 0 : atan2(Pt(), fZ); }
 
    /**
        pseudorapidity

--- a/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
@@ -154,7 +154,7 @@ public :
    /**
       Energy
     */
-   Scalar E() const { return sqrt(E2()); }
+   Scalar E() const { return std::sqrt(E2()); }
 
    Scalar T() const { return E();}
 
@@ -166,7 +166,7 @@ public :
    /**
       magnitude of spatial components (magnitude of 3-momentum)
    */
-   Scalar P() const { return sqrt(P2()); }
+   Scalar P() const { return std::sqrt(P2()); }
    Scalar R() const { return P(); }
 
    /**
@@ -198,7 +198,7 @@ public :
    /**
       Transverse spatial component (P_perp or rho)
    */
-   Scalar Pt() const { return sqrt(Perp2()); }
+   Scalar Pt() const { return std::sqrt(Perp2()); }
    Scalar Perp() const { return Pt();}
    Scalar Rho()  const { return Pt();}
 
@@ -213,11 +213,11 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return sqrt(mm);
+         return std::sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzM4D::Mt() - Tachyonic:\n"
                            "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -sqrt(-mm);
+         return -std::sqrt(-mm);
       }
    }
 
@@ -235,7 +235,7 @@ public :
    */
    Scalar Et() const {
       const Scalar etet = Et2();
-      return sqrt(etet);
+      return std::sqrt(etet);
    }
 
    /**

--- a/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
@@ -154,7 +154,7 @@ public :
    /**
       Energy
     */
-   Scalar E() const { return std::sqrt(E2()); }
+   Scalar E() const { using std::sqrt; return sqrt(E2()); }
 
    Scalar T() const { return E();}
 
@@ -166,7 +166,7 @@ public :
    /**
       magnitude of spatial components (magnitude of 3-momentum)
    */
-   Scalar P() const { return std::sqrt(P2()); }
+   Scalar P() const { using std::sqrt; return sqrt(P2()); }
    Scalar R() const { return P(); }
 
    /**
@@ -198,7 +198,7 @@ public :
    /**
       Transverse spatial component (P_perp or rho)
    */
-   Scalar Pt() const { return std::sqrt(Perp2()); }
+   Scalar Pt() const { using std::sqrt; return sqrt(Perp2()); }
    Scalar Perp() const { return Pt();}
    Scalar Rho()  const { return Pt();}
 
@@ -213,11 +213,13 @@ public :
    Scalar Mt() const {
       const Scalar mm = Mt2();
       if (mm >= 0) {
-         return std::sqrt(mm);
+         using std::sqrt;
+         return sqrt(mm);
       } else {
          GenVector::Throw ("PxPyPzM4D::Mt() - Tachyonic:\n"
                            "    Pz^2 > E^2 so the transverse mass would be imaginary");
-         return -std::sqrt(-mm);
+         using std::sqrt;
+         return -sqrt(-mm);
       }
    }
 
@@ -235,7 +237,8 @@ public :
    */
    Scalar Et() const {
       const Scalar etet = Et2();
-      return std::sqrt(etet);
+      using std::sqrt;
+      return sqrt(etet);
    }
 
    /**

--- a/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
+++ b/math/genvector/inc/Math/GenVector/PxPyPzM4D.h
@@ -244,12 +244,12 @@ public :
    /**
       azimuthal angle
    */
-   Scalar Phi() const { return (fX == 0.0 && fY == 0.0) ? 0.0 : atan2(fY, fX); }
+   Scalar Phi() const { using std::atan2; return (fX == 0.0 && fY == 0.0) ? 0.0 : atan2(fY, fX); }
 
    /**
       polar angle
    */
-   Scalar Theta() const { return (fX == 0.0 && fY == 0.0 && fZ == 0.0) ? 0 : atan2(Pt(), fZ); }
+   Scalar Theta() const { using std::atan2; return (fX == 0.0 && fY == 0.0 && fZ == 0.0) ? 0 : atan2(Pt(), fZ); }
 
    /**
        pseudorapidity

--- a/math/genvector/inc/Math/GenVector/RotationX.h
+++ b/math/genvector/inc/Math/GenVector/RotationX.h
@@ -94,13 +94,13 @@ public:
    /**
       Get the angle
    */
-   void GetAngle(Scalar &angle) const { angle = atan2(fSin, fCos); }
+   void GetAngle(Scalar &angle) const { using std::atan2; angle = atan2(fSin, fCos); }
    void GetComponents ( Scalar & angle ) const { GetAngle(angle); }
 
    /**
       Angle of rotation
    */
-   Scalar Angle() const { return atan2(fSin, fCos); }
+   Scalar Angle() const { using std::atan2; return atan2(fSin, fCos); }
 
    /**
       Sine or Cosine of the rotation angle

--- a/math/genvector/inc/Math/GenVector/RotationY.h
+++ b/math/genvector/inc/Math/GenVector/RotationY.h
@@ -94,13 +94,13 @@ public:
    /**
       Get the angle
    */
-   void GetAngle(Scalar &angle) const { angle = atan2(fSin, fCos); }
+   void GetAngle(Scalar &angle) const { using std::atan2; angle = atan2(fSin, fCos); }
    void GetComponents ( Scalar & angle ) const { GetAngle(angle); }
 
    /**
       Angle of rotation
    */
-   Scalar Angle() const { return atan2(fSin, fCos); }
+   Scalar Angle() const { using std::atan2; return atan2(fSin, fCos); }
 
    /**
       Sine or Cosine of the rotation angle

--- a/math/genvector/inc/Math/GenVector/RotationZ.h
+++ b/math/genvector/inc/Math/GenVector/RotationZ.h
@@ -94,13 +94,13 @@ public:
    /**
       Get the angle
    */
-   void GetAngle(Scalar &angle) const { angle = atan2(fSin, fCos); }
+   void GetAngle(Scalar &angle) const { using std::atan2; angle = atan2(fSin, fCos); }
    void GetComponents ( Scalar & angle ) const { GetAngle(angle); }
 
    /**
       Angle of rotation
    */
-   Scalar Angle() const { return atan2(fSin, fCos); }
+   Scalar Angle() const { using std::atan2; return atan2(fSin, fCos); }
 
    /**
       Sine or Cosine of the rotation angle

--- a/math/genvector/inc/Math/GenVector/VectorUtil.h
+++ b/math/genvector/inc/Math/GenVector/VectorUtil.h
@@ -277,8 +277,8 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateX(const Vector & v, double alpha) {
-            double sina = sin(alpha);
-            double cosa = cos(alpha);
+            double sina = std::sin(alpha);
+            double cosa = std::cos(alpha);
             double y2 = v.Y() * cosa - v.Z()*sina;
             double z2 = v.Z() * cosa + v.Y() * sina;
             Vector vrot;
@@ -294,8 +294,8 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateY(const Vector & v, double alpha) {
-            double sina = sin(alpha);
-            double cosa = cos(alpha);
+            double sina = std::sin(alpha);
+            double cosa = std::cos(alpha);
             double x2 = v.X() * cosa + v.Z() * sina;
             double z2 = v.Z() * cosa - v.X() * sina;
             Vector vrot;
@@ -311,8 +311,8 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateZ(const Vector & v, double alpha) {
-            double sina = sin(alpha);
-            double cosa = cos(alpha);
+            double sina = std::sin(alpha);
+            double cosa = std::cos(alpha);
             double x2 = v.X() * cosa - v.Y() * sina;
             double y2 = v.Y() * cosa + v.X() * sina;
             Vector vrot;

--- a/math/genvector/inc/Math/GenVector/VectorUtil.h
+++ b/math/genvector/inc/Math/GenVector/VectorUtil.h
@@ -108,7 +108,8 @@ namespace ROOT {
           */
          template <class Vector1, class Vector2>
          inline typename Vector1::Scalar DeltaR( const Vector1 & v1, const Vector2 & v2) {
-            return std::sqrt( DeltaR2(v1,v2) );
+            using std::sqrt;
+            return sqrt( DeltaR2(v1,v2) );
          }
 
 	/**
@@ -121,7 +122,8 @@ namespace ROOT {
           */
          template <class Vector1, class Vector2>
          inline typename Vector1::Scalar DeltaRapidityPhi( const Vector1 & v1, const Vector2 & v2) {
-            return std::sqrt( DeltaR2RapidityPhi(v1,v2) );
+            using std::sqrt;
+            return sqrt( DeltaR2RapidityPhi(v1,v2) );
          }
 
          /**
@@ -144,7 +146,8 @@ namespace ROOT {
                arg = 0.0;
             }else{
                double pdot = v1.X()*v2.X() + v1.Y()*v2.Y() + v1.Z()*v2.Z();
-               arg = pdot/std::sqrt(ptot2);
+               using std::sqrt;
+               arg = pdot/sqrt(ptot2);
                if(arg >  1.0) arg =  1.0;
                if(arg < -1.0) arg = -1.0;
             }
@@ -162,7 +165,8 @@ namespace ROOT {
           */
          template <class Vector1, class Vector2>
          inline double Angle( const  Vector1 & v1, const Vector2 & v2) {
-            return std::acos( CosTheta(v1, v2) );
+            using std::acos;
+            return acos( CosTheta(v1, v2) );
          }
 
          /**
@@ -220,7 +224,8 @@ namespace ROOT {
           */
          template <class Vector1, class Vector2>
          inline double Perp( const  Vector1 & v, const Vector2 & u) {
-            return std::sqrt(Perp2(v,u) );
+            using std::sqrt;
+            return sqrt(Perp2(v,u) );
          }
 
 
@@ -245,7 +250,8 @@ namespace ROOT {
             Scalar yy = (v1.Y() + v2.Y() );
             Scalar zz = (v1.Z() + v2.Z() );
             Scalar mm2 = ee*ee - xx*xx - yy*yy - zz*zz;
-            return mm2 < 0.0 ? -std::sqrt(-mm2) : std::sqrt(mm2);
+            using std::sqrt;
+            return mm2 < 0.0 ? -sqrt(-mm2) : sqrt(mm2);
             //  PxPyPzE4D<double> q(xx,yy,zz,ee);
             //  return q.M();
             //return ( v1 + v2).mag();
@@ -277,8 +283,10 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateX(const Vector & v, double alpha) {
-            double sina = std::sin(alpha);
-            double cosa = std::cos(alpha);
+            using std::sin;
+            double sina = sin(alpha);
+            using std::cos;
+            double cosa = cos(alpha);
             double y2 = v.Y() * cosa - v.Z()*sina;
             double z2 = v.Z() * cosa + v.Y() * sina;
             Vector vrot;
@@ -294,8 +302,10 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateY(const Vector & v, double alpha) {
-            double sina = std::sin(alpha);
-            double cosa = std::cos(alpha);
+            using std::sin;
+            double sina = sin(alpha);
+            using std::cos;
+            double cosa = cos(alpha);
             double x2 = v.X() * cosa + v.Z() * sina;
             double z2 = v.Z() * cosa - v.X() * sina;
             Vector vrot;
@@ -311,8 +321,10 @@ namespace ROOT {
           */
          template <class Vector>
          Vector RotateZ(const Vector & v, double alpha) {
-            double sina = std::sin(alpha);
-            double cosa = std::cos(alpha);
+            using std::sin;
+            double sina = sin(alpha);
+            using std::cos;
+            double cosa = cos(alpha);
             double x2 = v.X() * cosa - v.Y() * sina;
             double y2 = v.Y() * cosa + v.X() * sina;
             Vector vrot;
@@ -359,7 +371,8 @@ namespace ROOT {
                GenVector::Throw ( "Beta Vector supplied to set Boost represents speed >= c");
                return LVector();
             }
-            double gamma = 1.0 / std::sqrt(1.0 - b2);
+            using std::sqrt;
+            double gamma = 1.0 / sqrt(1.0 - b2);
             double bp = bx*v.X() + by*v.Y() + bz*v.Z();
             double gamma2 = b2 > 0 ? (gamma - 1.0)/b2 : 0.0;
             double x2 = v.X() + gamma2*bp*bx + gamma*bx*v.T();
@@ -384,7 +397,8 @@ namespace ROOT {
                GenVector::Throw ("Beta Vector supplied to set Boost represents speed >= c");
                return LVector();
             }
-            T gamma = 1.0/ std::sqrt(1.0 - beta*beta);
+            using std::sqrt;
+            T gamma = 1.0/ sqrt(1.0 - beta*beta);
             typename LVector::Scalar x2 = gamma * v.X() + gamma * beta * v.T();
             typename LVector::Scalar t2 = gamma * beta * v.X() + gamma * v.T();
 
@@ -405,7 +419,8 @@ namespace ROOT {
                GenVector::Throw ("Beta Vector supplied to set Boost represents speed >= c");
                return LVector();
             }
-            double gamma = 1.0/ std::sqrt(1.0 - beta*beta);
+            using std::sqrt;
+            double gamma = 1.0/ sqrt(1.0 - beta*beta);
             double y2 = gamma * v.Y() + gamma * beta * v.T();
             double t2 = gamma * beta * v.Y() + gamma * v.T();
             LVector lv;
@@ -425,7 +440,8 @@ namespace ROOT {
                GenVector::Throw ( "Beta Vector supplied to set Boost represents speed >= c");
                return LVector();
             }
-            double gamma = 1.0/ std::sqrt(1.0 - beta*beta);
+            using std::sqrt;
+            double gamma = 1.0/ sqrt(1.0 - beta*beta);
             double z2 = gamma * v.Z() + gamma * beta * v.T();
             double t2 = gamma * beta * v.Z() + gamma * v.T();
             LVector lv;

--- a/math/genvector/inc/Math/GenVector/eta.h
+++ b/math/genvector/inc/Math/GenVector/eta.h
@@ -55,9 +55,11 @@ namespace ROOT {
               using std::fabs;
               if (std::fabs(z_scaled) < big_z_scaled) {
                  using std::sqrt;
+                 using std::log;
                  return log(z_scaled + std::sqrt(z_scaled * z_scaled + 1.0));
               } else {
                  // apply correction using first order Taylor expansion of sqrt
+                 using std::log;
                  return z > 0 ? log(2.0 * z_scaled + 0.5 / z_scaled) : -log(-2.0 * z_scaled);
               }
            }
@@ -89,6 +91,7 @@ namespace ROOT {
               return -r - etaMax<Scalar>();
            }
            else {
+              using std::log;
               return -log(tanThetaOver2);
            }
 

--- a/math/genvector/inc/Math/GenVector/eta.h
+++ b/math/genvector/inc/Math/GenVector/eta.h
@@ -53,7 +53,7 @@ namespace ROOT {
 
               Scalar z_scaled = z/rho;
               if (std::fabs(z_scaled) < big_z_scaled) {
-                 return log(z_scaled + sqrt(z_scaled * z_scaled + 1.0));
+                 return log(z_scaled + std::sqrt(z_scaled * z_scaled + 1.0));
               } else {
                  // apply correction using first order Taylor expansion of sqrt
                  return z > 0 ? log(2.0 * z_scaled + 0.5 / z_scaled) : -log(-2.0 * z_scaled);

--- a/math/genvector/inc/Math/GenVector/eta.h
+++ b/math/genvector/inc/Math/GenVector/eta.h
@@ -52,7 +52,9 @@ namespace ROOT {
               static const Scalar big_z_scaled = pow(std::numeric_limits<Scalar>::epsilon(), static_cast<Scalar>(-.25));
 
               Scalar z_scaled = z/rho;
+              using std::fabs;
               if (std::fabs(z_scaled) < big_z_scaled) {
+                 using std::sqrt;
                  return log(z_scaled + std::sqrt(z_scaled * z_scaled + 1.0));
               } else {
                  // apply correction using first order Taylor expansion of sqrt

--- a/math/genvector/inc/Math/GenVector/etaMax.h
+++ b/math/genvector/inc/Math/GenVector/etaMax.h
@@ -34,9 +34,10 @@ namespace ROOT {
      */
     inline
     long double etaMax_impl() {
-      return std::log ( std::numeric_limits<long double>::max()/256.0l ) -
-             std::log ( std::numeric_limits<long double>::denorm_min()*256.0l )
-             + 16.0 * std::log(2.0);
+      using std::log;
+      return log ( std::numeric_limits<long double>::max()/256.0l ) -
+             log ( std::numeric_limits<long double>::denorm_min()*256.0l )
+             + 16.0 * log(2.0);
     // Actual usage of etaMax() simply returns the number 22756, which is
     // the answer this would supply, rounded to a higher integer.
     }

